src/deprecated.jl

```diff
@@ -1,3 +1,162 @@
-Base.@deprecate_binding AbstractContext Assertion
-@deprecate Context(; limit=1.0, kwargs...) Threshold(limit)
-@deprecate WeightedContext(wv; limit=1.0, kwargs...) Threshold(limit; weights=wv)
+# Introduced in 0.6
+Base.@deprecate_binding(
+    AbstractContext,
+    Assertion,
+    false,
+    "Imputation contexts have been replaced with independent Impute.Assertion types.",
+)
+Base.@deprecate Context(; limit=1.0, kwargs...) Threshold(limit) false
+Base.@deprecate WeightedContext(wv; limit=1.0, kwargs...) Threshold(limit; weights=wv) false
+
+# A couple utility methods to avoid messing up var and obs dimensions
+# NOTE: We aren't deprecating these as they were always internal function that weren't
+# intended for public use.
+obsdim(dims::Int) = dims
+vardim(dims::Int) = dims == 1 ? 2 : 1
+
+function obswise(data::AbstractMatrix; dims=1)
+    return (selectdim(data, obsdim(dims), i) for i in axes(data, obsdim(dims)))
+end
+
+function varwise(data::AbstractMatrix; dims=2)
+    return (selectdim(data, vardim(dims), i) for i in axes(data, vardim(dims)))
+end
+
+function filterobs(f::Function, data::AbstractMatrix; dims=1)
+    mask = [f(x) for x in obswise(data; dims=dims)]
+    return dims == 1 ? data[mask, :] : data[:, mask]
+end
+
+function filtervars(f::Function, data::AbstractMatrix; dims=2)
+    mask = [f(x) for x in varwise(data; dims=dims)]
```

Comment:
we should look at using filters ratehr than masks.
Save on allocation, but often less performant